src/Moq/Mock.cs

```diff
@@ -252,6 +252,11 @@ public void Verify()
 
 		private bool TryVerify(out MockException error)
 		{
+			foreach (Invocation invocation in this.MutableInvocations)
```

Comment:
There is some hidden complexity here. This returns an `IEnumerator<IInvocation>` that triggers one initial `lock` and enters a lock for each iteration. There's also a type cast here from `IInvocation` to `Invocation` so we get access to the internal `MarkAsVerifiedIf...` method.

This could be more efficient, but it's probably good enough for now.